neofetch@eson57/files/neofetch@eson57/neofetch@eson57.sh

```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+gnome-terminal -- neofetch
```

Comment:
There is a way to retrieve the logged in user's default terminal and the exec arguments to include. There are two gsettings (`exec` and `exec-arg`) under org.cinnamon.desktop.default-applications.terminal (/org/cinnamon/desktop/applications/terminal if you browse with dconf) that contain these values.

So you can retrieve the two values like:
```bash
EXEC=$(gsettings get org.cinnamon.desktop.default-applications.terminal exec)
EXECARG=$(gsettings get org.cinnamon.desktop.default-applications.terminal exec-arg)
```

I realize this is the same structure that was used in the run-speedtest Action so I apologize for the oversight there. Also, I tested both run-speedtest and this and neither were able to work for me without modification. More on that in the second comment.

Reply:
Last thing first! In gnome-terminal there is this setting under `Edit/Preferences/<your profile>/Command/When command exits`, that have to be set to 'Hold the terminal open'. I think default is 'Exit the terminal' and I haven't found a way to overrule this. The setting obviously overrules the `$SHELL` string.

Reply:
The rest what you suggest is a bit over my level of knowledge.
However, I found the below to identify default terminal and that seem to work, but I fail getting the script to run the neofetch command. So I need help with any of these solutions.

```
# Detect the default terminal emulator
TERMINAL_CMD=""
if command -v x-terminal-emulator &>/dev/null; then
    TERMINAL_CMD="x-terminal-emulator"
elif command -v gnome-terminal &>/dev/null; then
    TERMINAL_CMD="gnome-terminal"
elif command -v konsole &>/dev/null; then
    TERMINAL_CMD="konsole"
else
    echo "Error: No supported terminal emulator found."
    exit 1
fi
# Open a new terminal window and execute 'neofetch'
$TERMINAL_CMD -- neofetch
```

Reply:
> Last thing first! In gnome-terminal there is this setting under `Edit/Preferences/<your profile>/Command/When command exits`, that have to be set to 'Hold the terminal open'. I think default is 'Exit the terminal' and I haven't found a way to overrule this. The setting obviously overrules the `$SHELL` string.

Don't change this. People have their terminals configured for a reason. And again, this does nothing for the people who don't have the terminal installed or are using another as their preferred application.

Reply:
> The rest what you suggest is a bit over my level of knowledge. However, I found the below to identify default terminal and that seem to work, but I fail getting the script to run the neofetch command. So I need help with any of these solutions.
> 
> ```
> #!/usr/bin/bash
> # Detect the default terminal emulator
> TERMINAL_CMD=""
> if command -v x-terminal-emulator &>/dev/null; then
>     TERMINAL_CMD="x-terminal-emulator"
> elif command -v gnome-terminal &>/dev/null; then
>     TERMINAL_CMD="gnome-terminal"
> elif command -v konsole &>/dev/null; then
>     TERMINAL_CMD="konsole"
> else
>     echo "Error: No supported terminal emulator found."
>     exit 1
> fi
> # Open a new terminal window and execute 'neofetch'
> $TERMINAL_CMD -- neofetch
> ```

You can copy and paste the exact two lines I provided to fetch the values regardless of what the configuration is. The code that you've proposed here fails too many conditions otherwise.

Reply:
> The setting obviously overrules the `$SHELL` string.

This is the key thing. Maybe we should just skip these kinds of Actions for now. It's easy enough for most to open a terminal and type a single command just as easy as it is to install an Action. We could save some time and energy that would be better spent elsewhere.

Reply:
Sure, I can still use it as is internally.